packages/editor/src/editor/sidebar/fields/ColumnsCheckboxField.tsx

```diff
@@ -0,0 +1,88 @@
+import type { DataTable, DataTableColumn, Variable } from '@axonivy/form-editor-protocol';
+import { Checkbox, Field, Fieldset, Label, Message, type BrowserNode } from '@axonivy/ui-components';
+import { useMeta } from '../../../context/useMeta';
+import { useAppContext } from '../../../context/AppContext';
+import { createId, useData } from '../../../data/data';
+import { findAttributesOfType } from '../../../data/variable-tree-data';
+
+type ColumnsCheckboxFieldProps = {
+  label: string;
+  columns: DataTableColumn[];
+  onChange: (value: DataTableColumn[]) => void;
+};
+
+type CheckboxColumn = DataTableColumn & { selected: boolean };
+
+export const ColumnsCheckboxField = ({ label, columns, onChange }: ColumnsCheckboxFieldProps) => {
+  const { context } = useAppContext();
+  const { element } = useData();
+
+  const variableInfo = useMeta('meta/data/attributes', context, { types: {}, variables: [] }).data;
+  const attributesOfTableType = findAttributesOfType(variableInfo, element ? (element.config as unknown as DataTable).value : '');
+  const dataTableColumns = convertBrowserNodesToColumns(attributesOfTableType);
+
+  const selectableDataTableColumns: CheckboxColumn[] = dataTableColumns.map(column => ({
+    ...column,
+    selected: columns.some(col => col.config.value === column.config.value)
+  }));
+
+  const handleCheckboxChange = (e: boolean | 'indeterminate', column: CheckboxColumn) => {
+    if (e === true) {
+      const newColumn: DataTableColumn = {
+        config: {
+          header: column.config.header,
+          value: column.config.value
+        },
+        id: createId('DataTableColumn')
+      };
+      const newColumns = [...columns, newColumn];
+      onChange(newColumns);
+    } else if (e === false) {
+      const newColumns = columns.filter(col => !(col.config.value === column.config.value));
+      onChange(newColumns);
+    }
+  };
+
+  return dataTableColumns.length === 0 ? (
+    <Message variant='warning' message='Defined Object is not valid' />
+  ) : (
+    <Fieldset label={label}>
+      {selectableDataTableColumns.map(column => (
+        <Field direction='row' alignItems='center' gap={1} key={column.config.value}>
+          <Checkbox checked={column.selected} onCheckedChange={e => handleCheckboxChange(e, column)} />
+          <Label>{column.config.header}</Label>
+        </Field>
+      ))}
+    </Fieldset>
+  );
+};
+
+const convertBrowserNodesToColumns = (nodes: Array<BrowserNode<Variable>>): DataTableColumn[] => {
+  const columns: DataTableColumn[] = [];
+
+  const convertNode = (node: BrowserNode<Variable>) => {
+    if (!node.children || node.children.length === 0) {
+      columns.push({
+        id: createId('DataTableColumn'),
+        config: {
+          header: node.data?.attribute || '',
```

Comment:
|| is a logical operator, so the result of it is true or false. Isn't `??` the operator you're looking for?